sdk/examples/intkey_cxx/int_key_transaction_processor.cpp

```diff
@@ -0,0 +1,292 @@
+/* Copyright 2017 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------*/
+#include <iostream>
+#include <string>
+
+#include "log4cxx/logger.h"
+#include "log4cxx/basicconfigurator.h"
+
+#include "sawtooth/address_mapper.h"
+#include "sawtooth/transaction_processor.h"
+#include "json.hpp"
+
+using namespace log4cxx;
+
+using namespace nlohmann;
+
+static log4cxx::LoggerPtr logger(log4cxx::Logger::getLogger
+    ("sawtooth.IntKeyCXX"));
+
+static const std::string INTKEY_NAMESPACE = "intkey";
+
+// Custom Exception type defined to indicate any
+// failure in process a transaction.
+class InvalidTransaction: public std::runtime_error {
```

Comment:
Can this error be part of the SDK instead of intkey since it will be used by other processors?